File: backend/microservices/task-tracker/TaskTracker.Core/TaskTracker.Core.Application/Features/Challenges/Models/ChallengeDto.cs
using TaskTracker.Core.Domain.Challenges.Entities;

namespace TaskTracker.Core.Application.Features.Challenges.Models;

public sealed record ChallengeDto(Guid Id, string Name, string? Description, int UserId)
{
    public static ChallengeDto FromEntity(Challenge challenge)
    {
        return new ChallengeDto(challenge.Id, challenge.Name, challenge.Description, challenge.UserId);
    }
};

File: telegram-bot/models/challenge_dto.py
class ChallengeDto:
    id: int
    name: str
    description: str
    user_id: int

    def __init__(self, id: int, name: str, description: str, user_id: int):
        self.id = id
        self.name = name
        self.description = description
        self.user_id = user_id
        
    def from_json(json):
        return ChallengeDto(json['id'], json['name'], json['description'], json['userId'])
